src/main/java/com/example/hbs/service/UserService.java

```diff
@@ -50,20 +66,21 @@ public UserResponseDto addUser(UserRequestDto userRequestDto) {
     }
 
     public UserResponseDto updateUser(Long id, UserRequestDto userRequestDto) {
-        if (userRepository.findById(id).isPresent()) {
-            User user = userRepository.findById(id).get();
-            if (userRequestDto.getUserName() != null)
-                user.setUserName(userRequestDto.getUserName());
-            if (userRequestDto.getUserEmail() != null)
-                user.setUserEmail(userRequestDto.getUserEmail());
-            if (userRequestDto.getUserContact() != null)
-                user.setUserContact(userRequestDto.getUserContact());
-            if (userRequestDto.getUserRole() != null)
-                user.setUserRole(userRequestDto.getUserRole());
-            return userMapper.map(userRepository.save(user));
-        } else {
-            return null;
+        User user = userRepository.findById(id).orElse(null);
```

Comment:
you can throw exception here only,
`User user = userRepository.findById(id).orElseThrow(() -> new HbsException())`